app/src/main/java/de/danoeh/antennapod/ui/screen/playback/audio/AudioPlayerFragment.java

```diff
@@ -370,17 +374,44 @@ public void updatePosition(PlaybackPositionEvent event) {
         txtvPosition.setText(Converter.getDurationStringLong(currentPosition));
         txtvPosition.setContentDescription(getString(R.string.position,
                 Converter.getDurationStringLocalized(getContext(), currentPosition)));
-        showTimeLeft = UserPreferences.shouldShowRemainingTime();
+
+        int remainingTime = converter.convert(Math.max(event.getDuration() - event.getPosition(), 0));
+        final boolean endingThisEpisode = controller.isSleepTimerEndingThisEpisode(remainingTime);
+
+        showTimeLeft = UserPreferences.shouldShowRemainingTime() || controller.sleepTimerActive();
+        final String endingSymbol = endingThisEpisode ? "⏲" : "-";
+
         if (showTimeLeft) {
+            int remainingSleepTime = Math.toIntExact(controller.getSleepTimerTimeLeft().getMilisValue());
+            if (remainingSleepTime > 0) {
+                remainingTime = Math.min(remainingSleepTime, remainingTime);
+            }
+
             txtvLength.setContentDescription(getString(R.string.remaining_time,
                     Converter.getDurationStringLocalized(getContext(), remainingTime)));
-            txtvLength.setText(((remainingTime > 0) ? "-" : "") + Converter.getDurationStringLong(remainingTime));
+            txtvLength.setText(((remainingTime > 0) ? endingSymbol : "")
+                    + Converter.getDurationStringLong(remainingTime));
         } else {
             txtvLength.setContentDescription(getString(R.string.chapter_duration,
                     Converter.getDurationStringLocalized(getContext(), duration)));
             txtvLength.setText(Converter.getDurationStringLong(duration));
         }
 
+        if (controller.sleepTimerActive()) {
+            if (endingThisEpisode) {
+                // add a symbol to signal that playback is ending sometime this episode
+                final CharSequence text = txtvLength.getText();
+                txtvLength.setText(text + "");
+            }
+            txtvLength.setTextColor(
+                    MaterialColors.getColor(getContext(),
+                            android.R.attr.colorActivatedHighlight, Color.RED));
```

Comment:
Please move this to a separate PR. We are currently redesigning the player screen, so it is not yet clear whether we want to add this or not.